### PR TITLE
fix: cleanup client-landing-zone

### LINKED ISSUE #551: docs: client-landing-zone - security control comments
comments are formatted differently throughout the package

they should be revised

### LINKED ISSUE #552: fix: client-landing-zone - fw rule default ingress deny, no logging
the hidden default ingress deny firewall rule doesn't have logging enabled

a new one should be created in `https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit/blob/main/solutions/client-landing-zone/client-folder/standard/applications-infrastructure/host-project/network/firewall.yaml` with logging enabled

### LINKED ISSUE #553: fix: client-landing-zone - fw rule default egress deny, incorrect source range
the source range for default egress deny is hardcoded to 10.0.0.0/0

it should be set to 0.0.0.0/0 instead

### LINKED ISSUE #554: fix: client-landing-zone - fw rule inconsistent setter name
the setter named `firewall-egress-allow-all-internal` is also used by ingress rules

it should be renamed to generically represent the internal IP ranges (i.e. `firewall-internal-ip-ranges`)

### LINKED ISSUE #555: fix: client-landing-zone - VPC network routing mode
There is an issue with communication between region when vpc routing mode is set to regional.

Tests have been conducted to confirm that no impact to any existing functionnality will be affected if routing mode is changed to global.

Once router and nat exist in each region, the traffic originating from that region is sent to the cloud nat in the same region.
`https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit/blob/main/solutions/client-landing-zone/client-folder/standard/applications-infrastructure/host-project/network/vpc.yaml`

### PR DESCRIPTION
**Pull request by alaincormier-ssc**
general comments and setters cleanup for `client-landing-zone` package, as well as issues below:

closes #551
- standardize security control comments

closes #552
- add new default ingress deny rule for logging purposes

closes #553 
- edit default egress deny source range

closes #554
- rename `firewall-egress-allow-all-internal` setter to `firewall-internal-ip-ranges`

closes #555
- change vpc routing mode

### CODE REVIEW

**Review by fmichaelobrien — APPROVED** [review]
/lgtm

File: solutions/client-landing-zone/client-folder/standard/applications-infrastructure/host-project/network/vpc.yaml [hunk 1]
```diff
@@ -25,6 +25,6 @@ metadata:
 spec:
   resourceID: global-standard-vpc
   description: VPC for internet facing application
-  routingMode: REGIONAL
+  routingMode: GLOBAL
```
Comment (fmichaelobrien) [inline solutions/client-landing-zone/client-folder/standard/applications-infrastructure/host-project/network/vpc.yaml:28]: Good one - global will work better for any BGP routing

File: solutions/client-landing-zone/client-folder/standard/firewall-policy/rules/network-isolation.yaml [hunk 2]
```diff
@@ -22,10 +23,11 @@ metadata:
     config.kubernetes.io/depends-on: compute.cnrm.cloud.google.com/namespaces/client-name-networking/ComputeFirewallPolicy/client-name-standard-fwpol # kpt-set: compute.cnrm.cloud.google.com/namespaces/${client-name}-networking/ComputeFirewallPolicy/${client-name}-standard-fwpol
 spec:
   action: "deny"
-  description: "isolate nonp"
+  description: "Isolate non-protected subnet so it denies ingress traffic from pbmm subnet"
```
Comment (fmichaelobrien) [inline solutions/client-landing-zone/client-folder/standard/firewall-policy/rules/network-isolation.yaml:26]: thanks for this - no more confusion with non-prod

File: solutions/client-landing-zone/setters.yaml [hunk 3]
```diff
@@ -104,12 +104,9 @@ data:
...
   # https://cloud.google.com/firewall/docs/firewall-policies-rule-details#geo-location-object
   # https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#Officially_assigned_code_elements
-  # The list below is Google's default, modify as required
+  # The list below MUST BE reconfigured. If the value stays XX it will fail reconciliation with an error for unrecognized country.
   denied-sanctioned-countries: |
-    - "CU"
-    - "IR"
-    - "KP"
-    - "SY"
+    - "XX"
```
Comment (fmichaelobrien) [inline solutions/client-landing-zone/setters.yaml:109]: good pattern for forced review of the sanction list! - to force an error via purposely busted list

**Review by fmichaelobrien — DISMISSED** [review]
More full review of the entire PR
/lgtm

File: solutions/client-landing-zone/client-folder/firewall-policy/rules/lb-health-checks.yaml [hunk 4]
```diff
@@ -22,11 +23,12 @@ metadata:
     config.kubernetes.io/depends-on: compute.cnrm.cloud.google.com/namespaces/client-name-networking/ComputeFirewallPolicy/client-name-client-folder-fwpol # kpt-set: compute.cnrm.cloud.google.com/namespaces/${client-name}-networking/ComputeFirewallPolicy/${client-name}-client-folder-fwpol
 spec:
   action: "goto_next"
-  description: "Goto next for LB health checks"
+  description: "Delegate to host project, ingress LB health checks traffic (firewall) from IP ranges 35.191.0.0/16, 130.211.0.0/22  to VPC resources in shared VPC network"
```
Comment (lucstjean-ssc) [inline solutions/client-landing-zone/client-folder/firewall-policy/rules/lb-health-checks.yaml:26]: Not sure if a linter will catch it, but there's a double space between the IP and to: `130.211.0.0/22  to VPC`

**Review by cartyc — APPROVED** [review]
lgtm.

File: solutions/client-landing-zone/README.md [hunk 5]
```diff
@@ -18,13 +18,13 @@ Package to create a client's folder hierarchy, logging resources and a network h
...
-| client-name                          | client1                                                                                                                                                                                                                            | str   |   185 |
-| denied-sanctioned-countries          | ["CU", "IR", "KP", "SY"]                                                                                                                                                                                                           | array |     1 |
+| client-name                          | client1                                                                                                                                                                                                                            | str   |   187 |
+| denied-sanctioned-countries          | ["XX"]                                                                                                                                                                                                                             | array |     1 |
```
Comment (fmichaelobrien) [inline solutions/client-landing-zone/README.md:22]: I like the force-modify pattern

**Review by fmichaelobrien — APPROVED** [review]
/lgtm

**Review by lucstjean-ssc — APPROVED** [review]
/lgtm

**Pull request merged**